Drive

RahulBansal1112 opened the pull request.

RahulBansal1112 committed:
added teleop, auto, and odometry from offseason

RahulBansal1112 committed:
added teleop, auto, and odometry from offseason

RahulBansal1112 committed:
added DriveFSMSystem class

RahulBansal1112 committed:
added DriveFSMSystem class

RahulBansal1112 committed:
added 2022 navx vendor library

RahulBansal1112 committed:
added 2022 navx vendor library

MG-05 committed:
updates from the lab from Jan. 12. Autonomous does not work. The robot continues to run indefinitely.

MG-05 committed:
updates from the lab from Jan. 12. Autonomous does not work. The robot continues to run indefinitely.

Penguins478 committed:
implemented turning in place, speed modes, and acceleration

Penguins478 committed:
implemented turning in place, speed modes, and acceleration

RahulBansal1112 committed:
Merge remote-tracking branch 'origin/AutoTest' into Drive

RahulBansal1112 committed:
fixed import DriveFSM import statement in Robot.java

RahulBansal1112 committed:
fixed import DriveFSM import statement in Robot.java

RahulBansal1112 committed:
tested odometry calculations

RahulBansal1112 committed:
tested odometry calculations

MG-05 committed:
updated with revisions based on previous test

MG-05 committed:
updated with revisions based on previous test

Penguins478 committed:
added nonlinear driver input to output speeds

Penguins478 committed:
added nonlinear driver input to output speeds

RahulBansal1112 committed:
Merge remote-tracking branch 'origin/AutoTest' into Drive

RahulBansal1112 committed:
fixed bugs with the code and merged teleop and autoTest onto Drive

RahulBansal1112 committed:
fixed bugs with the code and merged teleop and autoTest onto Drive

RahulBansal1112 committed:
auto now goes forward and stops

RahulBansal1112 committed:
auto now goes forward and stops

MG-05 committed:
Got speed modes to work and turning in place is corrected

MG-05 committed:
Got speed modes to work and turning in place is corrected

MG-05 committed:
Added an excelling method of turning while driving

MG-05 committed:
Added an excelling method of turning while driving

RahulBansal1112 committed:
Merge remote-tracking branch 'origin/AutoTest' into Drive

mukundkk committed:
Added initial utility classes for trajectory code

mukundkk committed:
Added initial utility classes for trajectory code

RahulBansal1112 committed:
cleaned up code and added acceleration

RahulBansal1112 committed:
cleaned up code and added acceleration

RahulBansal1112 committed:
added acceleration, switching driving directions, adjusted driver input -> output algo, changed gyro

RahulBansal1112 committed:
added acceleration, switching driving directions, adjusted driver input -> output algo, changed gyro

RahulBansal1112 committed:
cleaned up import statements

RahulBansal1112 committed:
cleaned up import statements

MG-05 committed:
robot moves forward and back for a certain distance and robot accelerates and de-accelerates when moving forward.

MG-05 committed:
robot moves forward and back for a certain distance and robot accelerates and de-accelerates when moving forward.

mukundkk committed:
Add comments for reminders and TODOs in PurePursuit class

mukundkk committed:
Add comments for reminders and TODOs in PurePursuit class

mukundkk committed:
Fix naming error in loop

mukundkk committed:
Fix naming error in loop

RahulBansal1112 committed:
Merge remote-tracking branch 'origin/AutoTest' into Drive

RahulBansal1112 committed:
now compiles properly

RahulBansal1112 committed:
now compiles properly

RahulBansal1112 committed:
cleaned up code

RahulBansal1112 committed:
cleaned up code

RahulBansal1112 committed:
merged changes into Drive

RahulBansal1112 committed:
fixed build issues

RahulBansal1112 committed:
fixed build issues

RahulBansal1112 committed:
further cleaned up drivefsmsystem code

RahulBansal1112 committed:
further cleaned up drivefsmsystem code

RahulBansal1112 committed:
fixed build issues

RahulBansal1112 committed:
fixed build issues

RahulBansal1112 committed:
changed Point class to hold doubles

RahulBansal1112 committed:
added kinematics file and added odometry to that file

mukundkk committed:
Finish basic point injection method + necessary changes in utility classes

mukundkk committed:
Refactor variable names for clarity

RahulBansal1112 committed:
added odometry to the kinematics file

RahulBansal1112 committed:
redid inverse kinematics calcs

RahulBansal1112 committed:
made odometry methods static

mukundkk committed:
Add closest point method & refactor kinematics to make access easier

mukundkk committed:
Fix build errors

mukundkk committed:
Add method to calculate curvature of arc

Penguins478 committed:
added tank drive code

Penguins478 committed:
added tank drive code

Penguins478 committed:
added tank drive code

Penguins478 committed:
added tank drive code

Penguins478 committed:
added tank drive code

RahulBansal1112 committed:
tested tank drive and gyro

RahulBansal1112 committed:
tested tank drive and gyro

RahulBansal1112 committed:
fixed build errors

RahulBansal1112 committed:
variety of changes to clean up the code

RahulBansal1112 committed:
variety of changes to clean up the code

RahulBansal1112 committed:
changes start state to turning and fixed turning code

RahulBansal1112 committed:
changes start state to turning and fixed turning code

RahulBansal1112 committed:
completed r2 auto

RahulBansal1112 committed:
completed r2 auto

RahulBansal1112 committed:
tested inverse kinematics

RahulBansal1112 committed:
changes teleop to tank drive

RahulBansal1112 committed:
rebase Drive onto main

RahulBansal1112 committed:
fixed compiling error

RahulBansal1112 committed:
fixed rebase issue

RahulBansal1112 committed:
fixed most of the checkstyle errors

RahulBansal1112 committed:
fixed some merge issues

MG-05 committed:
made checkstyle fixes

MG-05 committed:
fixed checkstyle issues

RahulBansal1112 committed:
added path distance and angles as constants

RahulBansal1112 committed:
fixed teleopinput checkstyle errors

RahulBansal1112 committed:
rebased PurePursuit on Drive

RahulBansal1112 committed:
fixed issue that calculated targetAngle in radians instead of degrees in inverse kinematics calcs

RahulBansal1112 committed:
tested pure pursuit

RahulBansal1112 committed:
inverse kinematics: robot now turns if the target point isn't within a certain angle in front of it

Penguins478 committed:
changed straight drive constant

tsoim commented:
try rebasing on main first then pushing / creating the pull request? seems like you're trying to bring in changes that have already been merged

MG-05 committed:
adjusted constants so Path 1 for R2 Auto works

AG6GR committed:
Squash changes from PurePursuit/Drive branch

This commit is equivalent to the changes from the Drive or
PurePursuit branch applied on top of main.

RahulBansal1112 committed:
merged TeleOp into Drive

RahulBansal1112 committed:
Merge remote-tracking branch 'origin/AutoTest' into Drive

RahulBansal1112 committed:
fixed minor issues with r3 auto edge cases

RahulBansal1112 committed:
added pure pursuit circle

AG6GR commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -23,17 +23,18 @@
-	public static final double RUN_4_BACK_TO_TARMAC_DIST = -83.577;
-	public static final double RUN_4_TURN_TO_HUB_ANGLE = 69.125;
-	public static final double RUN_4_BACK_TO_HUB_DIST = -23.128;
+	public static final double RUN_4_LEAVE_TARMAC_DIST = 55.699;
```
What are the units for these values?

AG6GR commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -19,7 +19,7 @@
 	// Systems
 	private DriveFSMSystem driveFsmSystem;
-	private BallHandlingFSM ballSystem;
+	// private BallHandlingFSM ballSystem;
```
Let's leave all systems enabled on main branch, just comment for local testing.

AG6GR commented on src/main/java/frc/robot/trajectory/Kinematics.java:
```diff
@@ -0,0 +1,126 @@
+	public static Point inversekinematics(double gyroHeading, Point robotPos, Point targetPos) {
+		Point newtargetPos = new Point(targetPos.getX() - robotPos.getX(), targetPos.getY() - robotPos.getY());
+
+		double mRC = Math.tan(Math.toRadians(gyroHeading));
```
Can you rename these single letter variables, or add comments on what these represent? Also the units.

AG6GR commented on src/main/java/frc/robot/trajectory/Point.java:
```diff
@@ -0,0 +1,45 @@
+package frc.robot.trajectory;
+
+public class Point {
```
Consider using the Translation2D class provided by WPILib https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Translation2d.html

AG6GR commented on src/main/java/frc/robot/trajectory/Vector.java:
```diff
@@ -0,0 +1,48 @@
+package frc.robot.trajectory;
+
+public class Vector {
```
Same as with Point, could probably be replaced with Translation2D or Pose. https://docs.wpilib.org/en/stable/docs/software/advanced-controls/geometry/pose.html

AG6GR commented on src/main/java/frc/robot/trajectory/PurePursuit.java:
```diff
@@ -0,0 +1,107 @@
+		return lookaheadPoint;
+	}
+
+	// private double calculateCurvature() {
```
Clean comments plz :)

tsoim replied:
consider git stash if there are things you want to save but not commit to main

https://www.atlassian.com/git/tutorials/saving-changes/git-stash

AG6GR commented on src/main/java/frc/robot/trajectory/PurePursuit.java:
```diff
@@ -0,0 +1,107 @@
+	private static final double SPACING = 6.0;
+
+
+	public PurePursuit(ArrayList<Point> initialPoints, DriveFSMSystem fsmSystem) {
```
I'm not a fan of passing in fsmSystem to the constructor. DriveFSMSystem owns the PurePursuit instance, not the other way around. As far as I can tell, we only use fsmSystem to query values like getRobotPosArc and getheading. These should be passed in the method parameters when DriveFSMSystem calls the method.

MG-05 committed:
Added autorun constants for all 3 runs and changed .json file to run on old robot

RahulBansal1112 committed:
merged PurePursuit into Drive

tsoim commented on .wpilib/wpilib_preferences.json:
```diff
@@ -2,5 +2,5 @@
     "currentLanguage": "java",
     "projectYear": "2022",
-    "teamNumber": 2473
+    "teamNumber": 2475
```
?

RahulBansal1112 replied:
We changed the old robot to have team number 2475 (so the ip for the router is different). So to test code on that robot, we needed to change the team number to 2475 (prob shouldn't have pushed that change to Drive though).

tsoim commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -256,7 +308,9 @@ private void handleForwardOrBackwardState(TeleopInput input,
 
-		// Try This Next Time:
+		// Checks if either the encoder value is equal to the inches required (reached destination)
+		// or checks if the robot has hit a wall (encoder value is not chaning but motor power is
```
this case is not reflected in code? also typo for "chaning" -> "changing"

tsoim commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -256,7 +308,9 @@ private void handleForwardOrBackwardState(TeleopInput input,
+	// not zero)
 		if ((inches > 0 && error < Constants.ERR_THRESHOLD_STRAIGHT_IN)
 			|| (inches < 0 && error > -Constants.ERR_THRESHOLD_STRAIGHT_IN)) {
 			System.out.println("im here");
```
debug print statements shouldn't be included in PR, or at least make them more descriptive

RahulBansal1112 replied:
Yea we'll remove them

tsoim commented on src/main/java/frc/robot/systems/DriveFSMSystem.java:
```diff
@@ -375,46 +432,109 @@ private void handleTeleOpState(TeleopInput input) {
+		frontLeftMotor.set(-motorSpeeds.getX() / 5);
+		frontRightMotor.set(motorSpeeds.getY() / 5);
+		backLeftMotor.set(-motorSpeeds.getX() / 5);
+		backRightMotor.set(motorSpeeds.getY() / 5);
```
magic num `5`?

RahulBansal1112 replied:
We will be cleaning up the code soon (we've been procrastinating on fixing the current checkstyle errors).

tsoim replied:
as a mentor I'm obligated to say that you should follow these best practices as you're coding, not after the fact :P please try to clean the code up before putting up a PR next time, it'll help reviewers focus on the actual code and not be confused/distracted by misc stuff :)

mukundkk committed:
Update PurePursuit class to not take fsmSystem in constructor

Edit closest point method to account for equidistant points and fix
some checkstyle errors.

RahulBansal1112 committed:
change to 2 motors

Penguins478 committed:
added endgame alignment

RahulBansal1112 committed:
combine two pure pursuit paths

mukundkk committed:
Revert to original points and increase speed

mukundkk committed:
Add gyro turn to end of path

Finish r3 auto path and increase speed as well.

RahulBansal1112 committed:
Add javadoc comments and fix checkstyle errors

RahulBansal1112 committed:
Add new Pure Pursuit constants

mukundkk committed:
Add javadoc comments for some methods

RahulBansal1112 committed:
Test auto path

RahulBansal1112 committed:
Test auto run

RahulBansal1112 committed:
Merge PurePursuit into Drive

RahulBansal1112 committed:
Fix merge error with Drive

Penguins478 committed:
Replaced magic numbers with constants

Penguins478 committed:
Fixed checkstyle errors

mukundkk committed:
Fix checkstyle errors in trajectory code

RahulBansal1112 committed:
Merge PurePursuit into Drive

RahulBansal1112 committed:
Add AutoPaths file to store paths

RahulBansal1112 committed:
Merge Drive into main

RahulBansal1112 committed:
Add back in ball fsm to Robot.java

RahulBansal1112 committed:
Suppress MagicNumber check for AutoPaths.java

RahulBansal1112 committed:
Supress MagicNumber for AutoPaths

AG6GR commented on src/main/java/frc/robot/trajectory/Kinematics.java:
```diff
@@ -0,0 +1,192 @@
+				/ (2.0 * (mRC - mRT < 0 ? -1 : 1) * Constants.ZERO_THRESHOLD));
+			//if the heading is one way and the target point is 180 deg
+			//in the other direction, set the center of the circle at ~(0, 0)
+			if (gyroHeading > 270 && gyroHeading < 90 & newtargetPos.getX() < 0) {
```
gyroHeading > 270 && gyroHeading < 90 is always false?

RahulBansal1112 committed:
Add suppressions file

RahulBansal1112 committed:
Fix checkstyle error

RahulBansal1112 committed:
Add SuppressionSingleFilter to filter out magic number check

RahulBansal1112 committed:
Remove suppressions.xml file
